Yosemite/Yosemite/Stores/CustomerStore.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import Networking
+import Storage
+
+// MARK: - CustomerStore
+//
+public class CustomerStore: Store {
+    private let remote: CustomerRemote
+
+    private lazy var sharedDerivedStorage: StorageType = {
+        return storageManager.newDerivedStorage()
+    }()
+
+    public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
+        self.remote = CustomerRemote(network: network)
+        super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
+    }
+
+    /// Registers for supported Actions.
+    ///
+    override public func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: CustomerAction.self)
+    }
+
+    /// Receives and executes Actions.
+    ///
+    override public func onAction(_ action: Action) {
+        guard let action = action as? CustomerAction else {
+            assertionFailure("CustomerStore received an unsupported action")
+            return
+        }
+
+        switch action {
+        case .synchronizeAllCustomers(let siteID, let completion):
+            synchronizeAllCustomers(siteID: siteID, completion: completion)
+        }
+    }
+}
+
+// MARK: - Services
+//
+private extension CustomerStore {
+
+    /// Retrieves all of the customers associated with a given Site ID (if any!).
+    ///
+    func synchronizeAllCustomers(siteID: Int64, fromPageNumber: Int = CustomerRemote.Default.pageNumber, completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
Because it calls for many pages recursively and collecting all of them in separate array to return in the end seems like overkill to me. Pagination on itself implies possibility of huge number of items. I think UI layer should do it on its own with `ResultsController`.

Reply:
Oh yes sounds good, thanks for clarifying. 🙇